src/structure/pdb.jl

```
@@ -0,0 +1,274 @@
+            parsestrict(line, (39,46), Float64, "Could not read y coordinate", line_number),
+            parsestrict(line, (47,54), Float64, "Could not read z coordinate", line_number)
+        ],
+        parselenient(line, (55,60), Float64, 1.0),
+        parselenient(line, (61,66), Float64, 0.0),
+        strip(parselenient(line, (77,78), ASCIIString, "")),
+        strip(parselenient(line, (79,80), ASCIIString, ""))
+    )
+end
+
+
+"Parse columns from a line and return the value or throw a `PDBParseError`."
+function parsestrict(line::ASCIIString,
+                    cols::Tuple{Int, Int},
+                    out_type::Type,
+                    error_message::ASCIIString,
+                    line_number::Int)
+    try
+        return parsevalue(line, cols, out_type)
+    catch
+        throw(PDBParseError(error_message, line_number, line))
+    end
+end
+
+
+"Parse columns from a line and return the value or a default value."
+function parselenient(line::ASCIIString,
+                    cols::Tuple{Int, Int},
+                    out_type::Type,
+                    default)
+    try
+        return parsevalue(line, cols, out_type)
+    catch
+        return default
+    end
+end
+
+
+"Parse columns from a line."
+function parsevalue(line::ASCIIString, cols::Tuple{Int, Int}, out_type::Type)
+    try
+        if out_type == Int
+            return parse(Int, line[cols[1]:cols[2]])
+        elseif out_type == Float64
+            return parse(Float64, line[cols[1]:cols[2]])
+        elseif out_type == ASCIIString
+            return line[cols[1]:cols[2]]
+        elseif out_type == Char
+            return line[cols[1]]
+        else
+            error()
+        end
+    catch
+        error("Could not parse to desired type")
+    end
+end
+
+
+"Form a string of a certain length from a value."
+function spacestring(val_in, new_length::Int)
+    string_out = string(val_in)
+    if length(string_out) > new_length
+        return string_out[1:new_length]
```

Is that safe to truncate `string_out`? It may be better to throw an error.